src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -288,6 +288,99 @@ protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes)
         public override string ToString() => "TakeWhile";
     }
 
+
+    /// <summary>
+    /// Take until
+    /// </summary>
+    /// <typeparam name="T">TBD</typeparam>
+    [InternalApi]
+    public sealed class TakeUntil<T> : SimpleLinearGraphStage<T>
+    {
+        #region Logic
+
+        private sealed class Logic : InAndOutGraphStageLogic
+        {
+            private readonly TakeUntil<T> _stage;
+            private readonly Decider _decider;
+
+            public Logic(TakeUntil<T> stage, Attributes inheritedAttributes) : base(stage.Shape)
+            {
+                _stage = stage;
+                var attr = inheritedAttributes.GetAttribute<ActorAttributes.SupervisionStrategy>(null);
+                _decider = attr != null ? attr.Decider : Deciders.StoppingDecider;
+
+                SetHandler(stage.Outlet, this);
+                SetHandler(stage.Inlet, this);
+            }
+
+            public override void OnPush()
+            {
+                try
+                {
+                    var element = Grab(_stage.Inlet);
+                    if (!_stage._predicate(element))
+                        Push(_stage.Outlet, element);
+                    else
+                    {
+                        if (_stage._inclusive)
```

Comment:
Shouldn't `TakeUntil` always be inclusive?

Reply:
Agreed - TakeUntil exclusive is basically TakeWhile exclusive with an inverted predicate